Modules/Filtering/DiffusionTensorImage/test/itkDiffusionTensor3DReconstructionImageFilterTest.cxx

```diff
@@ -118,7 +118,33 @@ itkDiffusionTensor3DReconstructionImageFilterTest(int argc, char * argv[])
       gradientDirection[2] = gradientDirections[i][2];
       tensorReconstructionFilter->AddGradientImage(gradientDirection, gradientImage);
       std::cout << "Gradient directions: " << gradientDirection << std::endl;
+
+      const TensorReconstructionImageFilterType::GradientDirectionType::element_type epsilon = 1e-3;
+      TensorReconstructionImageFilterType::GradientDirectionType                     output =
+        tensorReconstructionFilter->GetGradientDirection(i);
+      for (unsigned int j = 0; j < gradientDirection.size(); ++j)
+      {
+        TensorReconstructionImageFilterType::GradientDirectionType::element_type gradientDirectionComponent =
+          gradientDirection[j];
+        TensorReconstructionImageFilterType::GradientDirectionType::element_type outputComponent = output[j];
+        if (!itk::Math::FloatAlmostEqual(gradientDirectionComponent, outputComponent, 10, epsilon))
+        {
+          std::cerr.precision(static_cast<int>(itk::Math::abs(std::log10(epsilon))));
+          std::cerr << "Test failed!" << std::endl;
+          std::cerr << "Error in gradientDirection [" << i << "]"
+                    << "[" << j << "]" << std::endl;
+          std::cerr << "Expected value " << gradientDirectionComponent << std::endl;
+          std::cerr << " differs from " << outputComponent;
+          std::cerr << " by more than " << epsilon << std::endl;
+          return EXIT_FAILURE;
+        }
+      }
```

Comment:
Simplifying blocks like the one above, where a loop is required, through some testing macro would be great. Ideally, an `ITK_TEST_SET_GET_VALUE`-like being able to deal with ND arrays, including floating point comparisons, would make things easier in tests and more consistent.

Cross-referencing issue #1273.

Reply:
@blowekamp Does something like this exist in GTest? Or maybe some test helpers you wrote?

Reply:
This is the closes thing that I am aware of: https://github.com/InsightSoftwareConsortium/ITK/blob/dba14cdb283ba6c940a96e3f2199044c4dbff848/Modules/Core/TestKernel/include/itkGTestPredicate.h#L33-L34

Reply:
So I correctly remembered that you wrote something similar a few years ago 😄